src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java

```diff
@@ -36,31 +39,117 @@ public AnnotationConfigApplicationContext(String packageName) {
+        try {
+            registerBeansInContext(futureComponents);
+            autoWireBeans();
+        } catch (NoSuchFieldException  | NoSuchMethodException | InstantiationException |
+                 IllegalAccessException | InvocationTargetException e  ) {
+            LOGGER.error("Can't register beans in appliation context");
+            throw new RuntimeException(e);
+        } catch (NoUniqueBeanException e){
+            LOGGER.error("Can't register beans in appliation context");
+            throw new NoUniqueBeanException("MULTIPLE_BEANS_PRESENT");
+        } catch (NoSuchBeanException e){
+            LOGGER.error("Can't register beans in appliation context");
+            throw new NoSuchBeanException("BEAN_IS_NOT_PRESENT");
+        }
 
-        for (Class<?> bean : reflections.getTypesAnnotatedWith(Component.class)) {
-            String beanCustomName = bean.getAnnotation(Component.class).value();
-            String beanName = StringUtils.defaultIfEmpty(beanCustomName, WordUtils.uncapitalize(bean.getSimpleName()));
+    }
+
+    /**
+     * <h3>Registers components in application context</h3>
+     *
+     * @param classSet {@link Set<Class>}
+     */
+    private void registerBeansInContext(Set<Class<?>> classSet) throws NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        for (Class<?> beanClass : classSet) {
+            var constructor = beanClass.getConstructor();
+            var beanInstance = constructor.newInstance();
+            String beanName = resolveBeanName(beanClass);
             try {
-                context.put(beanName, bean.getDeclaredConstructor().newInstance());
-            } catch (InstantiationException | InvocationTargetException | IllegalAccessException |
-                     NoSuchMethodException e) {
+                context.put(beanName, beanInstance);
+            } catch (Exception e) {
                 LOGGER.error("Can't create an instance of {} class", beanName);
                 throw new RuntimeException(e);
             }
+
         }
     }
 
     /**
-     * Store bean in the context
+     * <h3>Injects Beans in Autowired field</h3>
+     */
+    private void autoWireBeans() throws IllegalAccessException, NoSuchFieldException {
+        for (Map.Entry<String, Object> entry : context.entrySet()) {
+            var beanInstance = entry.getValue();
+            injectBeanViaAutowiredAnnotation(beanInstance.getClass(), beanInstance);
+            injectBeanViaConstructor(beanInstance.getClass(), beanInstance);
+        }
+    }
+
+    /**
+     * <h3>Injects Beans via {@link Autowired} annotation</h3>
+     *
+     * @param beanClass    {@link Class}
+     * @param beanInstance {@link Object}
+     */
+    private <T> void injectBeanViaAutowiredAnnotation(Class<?> beanClass, T beanInstance) throws IllegalAccessException {
+        for (Field field : beanClass.getDeclaredFields()) {
+            field.setAccessible(true);
```

Comment:
We should place this field into if block. It is not necessary operation if condition false.